numpy/lib/function_base.py

```diff
@@ -1173,6 +1179,29 @@ def diff(a, n=1, axis=-1):
             "order must be non-negative but got " + repr(n))
 
     a = asanyarray(a)
+
+    combined = []
+    if prepend is not np._NoValue:
+        prepend = np.asanyarray(prepend)
+        if prepend.ndim == 0:
+            shape = list(a.shape)
+            shape[axis] = 1
+            prepend = np.broadcast_to(prepend, tuple(shape))
+        combined.append(prepend)
+
+    combined.append(a)
+
+    if append is not np._NoValue:
+        append = np.asanyarray(append)
+        if append.ndim == 0:
+            shape = list(a.shape)
+            shape[axis] = 1
+            append = np.broadcast_to(append, tuple(shape))
+        combined.append(append)
+
+    if len(combined) > 1:
+        a = np.concatenate(combined, axis)
+    
     nd = a.ndim
     axis = normalize_axis_index(axis, nd)
```

Comment:
This line needs to come before your additions

Reply:
can you please show me a short example?  I'll make this change and also add a test

Reply:
i assume its just a variation of this:
        assert_raises(np.AxisError, diff, x, axis=3)
        assert_raises(np.AxisError, diff, x, axis=-4)

Reply:
Yep, those tests, but with your new arguments